src/platform_impl/windows/dpi.rs

```diff
@@ -105,7 +105,7 @@ pub unsafe fn hwnd_dpi(hwnd: HWND) -> u32 {
         if unsafe { IsProcessDPIAware() } != false.into() {
             // If the process is DPI aware, then scaling must be handled by the application using
             // this DPI value.
-            unsafe { GetDeviceCaps(hdc, LOGPIXELSX) as u32 }
+            unsafe { GetDeviceCaps(hdc, LOGPIXELSX as i32) as u32 }
```

Comment:
We should ask for upstream clarification about these. At least in the `windows` crate where these are newtyped, the underlying ABI is "kept in sync" (IIRC by casting in the wrapper implementation).

Reply:
Given that casting small u32 to positive i32 have the same binary repr, it's safe to do it here.

In general, a lot of casts seems to be that way because sometimes they accept `enum` but `enum` itself is `i32` and not `u32`, and so on.

You can still ask though.

Reply:
Not saying that it's unsafe, just that it's inconvenient after these Windows crates already have all the type information they need; looks like it's all reserved for the high-level `windows` crate though.